server/autolink/lib_jira_test.go

```diff
@@ -80,43 +80,75 @@ var jiraTests = []linkTest{
 	}, {
 		"Not relinking",
 		autolink.Autolink{
-			Pattern:  "(MM)(-)(?P<jira_id>\\d+)",
-			Template: "[MM-$jira_id](https://mattermost.atlassian.net/browse/MM-$jira_id)",
+			Pattern:  "(?P<project_id>\\w+)(-)(?P<jira_id>\\d+)",
+			Template: "[${project_id}-${jira_id}](https://mattermost.atlassian.net/browse/${project_id}-${jira_id})",
```

Comment:
We aren't using this regex though right? From what I understand about the shorthand links, the Jira plugin will be filling in the project keys, and not use a project-agnostic `pattern`.

I believe this test would make it so something like  


> Here's my phone number-284836364


translates into:

```
Here's my phone [number-284836364](https://mattermost.atlassian.net/browse/number-284836364)
```

2/5 if this undesired behavior is a result of this regex structure, we shouldn't be using it in tests.

cc @hanzei